### PR TITLE
Upgrading the parser to version 6

### LINKED ISSUE #203: `FROM x@\<newline>sha256:...`: `expecting no '@' or a valid digest hash`
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Steps to reproduce the behavior

**Output of `hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.6.6-0-g9658652
```

**Dockerfile**

Case 1:
```Dockerfile
FROM python@\
sha256:b133ed1865bd7470239630c13e5834470f1c31673881924924643cefa461e9fb
...
```

Case 2:
```Dockerfile
FROM python\
@sha256:b133ed1865bd7470239630c13e5834470f1c31673881924924643cefa461e9fb
...
```

### Expected behavior

No warnings. The backslash in inserted to keep lines at no more than 80 characters. `docker build` (`Docker version 18.03.0-ce, build 0520e24`) parses `<before><backslash><newline><after>` correctly as `<before><after>`.

### Actual behavior

Case 1: `Dockerfile:1:13 unexpected '@' expecting no '@' or a valid digest hash (example: python@a3f42f2de)`
Case 2: `Dockerfile:1:13 unexpected "@" expecting space or "AS"`

It seems that `hadolint` parses `<before><backslash><newline><after>` as `<before><token separator><after>`. (Backslash handling is not part of text preprocessing, so `<before><after>` is not treated as a single token.)

### PR DESCRIPTION
**Pull request by lorenzo**
Uses the latest version of language-docker

fixes #203

### CODE REVIEW

**Comment by lorenzo** [comment]
@zemanlx do you have time to verify if after this change we can go back to the smaller docker image we were using before the UTF problems?

File: stack.yaml [hunk 1]
```diff
@@ -3,5 +3,5 @@ extra-package-dbs: []
 packages:
 - .
 extra-deps:
-- language-docker-5.0.0
-resolver: lts-10.9
+- language-docker-6.0.0
+resolver: lts-11.12
```
Comment (zemanlx) [inline stack.yaml]: Can you update it also in `.travis.yaml` please?

**Comment by zemanlx** [comment]
Yes I will test it.

for my reference - original issue https://github.com/hadolint/hadolint/issues/172

File: stack.yaml (hunk 1 above)
Comment (lorenzo) [reply stack.yaml]: Done

**Comment by zemanlx** [comment]
```bash
...
Step 9/11 : FROM busybox:1.27.2-glibc AS distro
1.27.2-glibc: Pulling from library/busybox
8da261274716: Pull complete 
Digest: sha256:8c8f261a462eead45ab8e610d3e8f7a1e4fd1cd9bed5bc0a0c386784ab105d8e
Status: Downloaded newer image for busybox:1.27.2-glibc
 ---> fe2d514cd106
Step 10/11 : COPY --from=builder /root/.local/bin/hadolint /bin/
 ---> 230233e36798
Step 11/11 : CMD ["/bin/hadolint", "-"]
 ---> Running in 3d593895b882
Removing intermediate container 3d593895b882
 ---> 8b60eb99e0fe
Successfully built 8b60eb99e0fe
Successfully tagged hadolint:v1.6.6-3-g52ba9a3-dirty

$ cat >Dockerfile.utf8
FROM debian:latest

LABEL maintainer="ö"

RUN echo "Hallo"

$ docker run --rm -i hadolint:v1.6.6-3-g52ba9a3-dirty < Dockerfile.utf8 
/dev/stdin:1 DL3007 Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag
```

so it looks good now :tada: I will make a PR for that.

**Pull request merged**